diff --git a/packages/excalidraw/components/App.tsx b/packages/excalidraw/components/App.tsx
index 0194a0ed..9097c45c 100644
--- a/packages/excalidraw/components/App.tsx
+++ b/packages/excalidraw/components/App.tsx
@@ -2399,9 +2399,12 @@ class App extends React.Component<AppProps, AppState> {
                             renderScrollbars={
                               this.props.renderScrollbars === true
                             }
                             hideDefaultRotateHandle={
                               this.props.hideDefaultRotateHandle === true
                             }
+                            getSelectionColorForElement={
+                              this.props.getSelectionColorForElement
+                            }
                             editorInterface={this.editorInterface}
                             renderInteractiveSceneCallback={
                               this.renderInteractiveSceneCallback
diff --git a/packages/excalidraw/components/canvases/InteractiveCanvas.tsx b/packages/excalidraw/components/canvases/InteractiveCanvas.tsx
index 936e5655..892a0f20 100644
--- a/packages/excalidraw/components/canvases/InteractiveCanvas.tsx
+++ b/packages/excalidraw/components/canvases/InteractiveCanvas.tsx
@@ -44,6 +44,7 @@ type InteractiveCanvasProps = {
   scale: number;
   appState: InteractiveCanvasAppState;
   renderScrollbars: boolean;
+  getSelectionColorForElement?: InteractiveCanvasRenderConfig["getSelectionColorForElement"];
   hideDefaultRotateHandle: boolean;
   editorInterface: EditorInterface;
   app: AppClassProperties;
@@ -159,6 +160,7 @@ const InteractiveCanvas = (props: InteractiveCanvasProps) => {
         remotePointerUserStates,
         selectionColor,
         renderScrollbars: props.renderScrollbars,
+        getSelectionColorForElement: props.getSelectionColorForElement,
         hideDefaultRotateHandle: props.hideDefaultRotateHandle,
         // NOTE not memoized on so we don't rerender on cursor move
         lastViewportPosition: props.app.lastViewportPosition,
@@ -289,7 +291,9 @@ const areEqual = (
     prevProps.visibleElements !== nextProps.visibleElements ||
     prevProps.selectedElements !== nextProps.selectedElements ||
     prevProps.renderScrollbars !== nextProps.renderScrollbars ||
-    prevProps.hideDefaultRotateHandle !== nextProps.hideDefaultRotateHandle
+    prevProps.hideDefaultRotateHandle !== nextProps.hideDefaultRotateHandle ||
+    prevProps.getSelectionColorForElement !==
+      nextProps.getSelectionColorForElement
   ) {
     return false;
   }
diff --git a/packages/excalidraw/index.tsx b/packages/excalidraw/index.tsx
index dfed2055..26d14177 100644
--- a/packages/excalidraw/index.tsx
+++ b/packages/excalidraw/index.tsx
@@ -104,6 +104,7 @@ const ExcalidrawBase = (props: ExcalidrawProps) => {
     aiEnabled,
     showDeprecatedFonts,
     renderScrollbars,
+    getSelectionColorForElement,
     hideDefaultRotateHandle,
     imageOptions,
   } = props;
@@ -222,6 +223,7 @@ const ExcalidrawBase = (props: ExcalidrawProps) => {
           aiEnabled={aiEnabled !== false}
           showDeprecatedFonts={showDeprecatedFonts}
           renderScrollbars={renderScrollbars}
+          getSelectionColorForElement={getSelectionColorForElement}
           hideDefaultRotateHandle={hideDefaultRotateHandle}
           imageOptions={normalizedImageOptions}
         >
diff --git a/packages/excalidraw/renderer/interactiveScene.ts b/packages/excalidraw/renderer/interactiveScene.ts
index 92560cea..44ce149e 100644
--- a/packages/excalidraw/renderer/interactiveScene.ts
+++ b/packages/excalidraw/renderer/interactiveScene.ts
@@ -1814,7 +1814,13 @@ const _renderInteractiveScene = ({
             locallySelectedIds.has(element.id) &&
             !isSelectedViaGroup(appState, element)
           ) {
-            selectionColors.push(selectionColor);
+            selectionColors.push(
+              renderConfig.getSelectionColorForElement?.(
+                element,
+                allElementsMap,
+                appState,
+              ) || selectionColor,
+            );
           }
           // remote users
           if (remoteClients) {
@@ -1869,7 +1875,7 @@ const _renderInteractiveScene = ({
           y2,
           selectionColors: groupElements.some((el) => el.locked)
             ? ["#ced4da"]
-            : ["#000"],
+            : [selectionColor],
           dashed: true,
           cx: x1 + (x2 - x1) / 2,
           cy: y1 + (y2 - y1) / 2,
diff --git a/packages/excalidraw/scene/types.ts b/packages/excalidraw/scene/types.ts
index 41977b7b..760ed8f5 100644
--- a/packages/excalidraw/scene/types.ts
+++ b/packages/excalidraw/scene/types.ts
@@ -67,6 +67,11 @@ export type InteractiveCanvasRenderConfig = {
   remotePointerUsernames: Map<SocketId, string>;
   remotePointerButton: Map<SocketId, string | undefined>;
   selectionColor: string;
+  getSelectionColorForElement?: (
+    element: NonDeletedExcalidrawElement,
+    allElementsMap: NonDeletedSceneElementsMap,
+    appState: InteractiveCanvasAppState,
+  ) => string | null | undefined;
   lastViewportPosition: { x: number; y: number };
   // extra options passed to the renderer
   // ---------------------------------------------------------------------------
diff --git a/packages/excalidraw/types.ts b/packages/excalidraw/types.ts
index a93a8530..6705af1e 100644
--- a/packages/excalidraw/types.ts
+++ b/packages/excalidraw/types.ts
@@ -33,6 +33,7 @@ import type {
   ExcalidrawNonSelectionElement,
   BindMode,
   ExcalidrawTextElement,
+  NonDeletedSceneElementsMap,
 } from "@excalidraw/element/types";
 
 import type {
@@ -686,6 +687,11 @@ export interface ExcalidrawProps {
   aiEnabled?: boolean;
   showDeprecatedFonts?: boolean;
   hideDefaultRotateHandle?: boolean;
+  getSelectionColorForElement?: (
+    element: NonDeletedExcalidrawElement,
+    allElementsMap: NonDeletedSceneElementsMap,
+    appState: InteractiveCanvasAppState,
+  ) => string | null | undefined;
   renderScrollbars?: boolean;
   /**
    * Called before exporting to a file.
